tasks/kmt.py

```diff
@@ -2371,3 +2375,70 @@ def flare(ctx: Context, dest_folder: Path | str | None = None, keep_uncompressed
 
     with tempfile.TemporaryDirectory() as tmpdir:
         flare_kmt_os(ctx, Path(tmpdir), dest_folder, keep_uncompressed_files)
+
+
+@task(help={"component": "The component to retry. If not provided, all components will be retried."})
+def retry_failed_pipeline(ctx: Context, pipeline_id: int, component: str | None = None):
+    """Retries all failed KMT tests in the given pipeline"""
+    info(f"[+] Retrieving jobs for pipeline {pipeline_id}")
+    kmt_pipeline = KMTPipeline(pipeline_id)
+    kmt_pipeline.retrieve_jobs()
+
+    jobs_to_retry: list[KMTTestRunJob] = []
+    failed_jobs = [j for j in kmt_pipeline.test_jobs if j.status == GitlabJobStatus.FAILED and not j.is_retried]
+    if len(failed_jobs) == 0:
+        info(f"[+] No failed tests found in pipeline {pipeline_id}. Checking dependency failures")
+
+        # Skipped tests can be skipped because of failed setup/dependency upload jobs, so check
+        # those too
+        skipped_jobs = [j for j in kmt_pipeline.test_jobs if j.status == GitlabJobStatus.SKIPPED]
+        for test in skipped_jobs:
+            if test.has_failed_dependencies:
+                jobs_to_retry.append(test)
+    else:
+        jobs_to_retry = failed_jobs
+
+    if component is not None:
+        jobs_to_retry = [j for j in jobs_to_retry if j.component == component]
+
+    if len(jobs_to_retry) == 0:
+        info(f"[+] No test jobs to retry in pipeline {pipeline_id}")
+        return
+
+    info(f"[+] Found the following candidates to retry in pipeline {pipeline_id}:")
```

Comment:
🥜 
```suggestion
    info(f"[+] Found {len(jobs_to_retry)} candidates to retry in pipeline {pipeline_id}(https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/{pipeline_id}):")
```
And delete the similar comment below